### ISSUE TITLE
MP4 encoded won't play in mobile

### ISSUE THREAD

**Issue by markginter31** [WWBN/AVideo#9263]
When I encode in MP4 - the video will play in all desktop browsers, but will not play in mobile.
When I encode in WEBM - the video will play on both desktop and mobile.

**To Reproduce**
1. enocode in MP4 - won't play
2. encode in WEBM - will play

**Expected behavior**
MP4 will play in both desktop and mobile

**Error Logs**
No errors appear in logs.  Running the latest version.

**Desktop (please complete the following information):**
Desktop -- Fedora 40 Firefox/Chrome
Mobile - Samsung S20 - Samsung Browser / Chrome

I'm not sure where to look to find why the MP4 doesn't play but the WEBM does.

**Comment by DanielnetoDotCom** [comment]
What exactly option are you using to encode?

automatic MP4 or are you forcing LOW, HD and SD?

**Comment by markginter31** [comment]
First I noticed when all the resolutions were checked in the encoder (I assume that's automatic)
Then I only chose 480, 720, and 1080 

Both times I got MP4's unplayable in mobile but playable on desktop.  

Is there a way in the mobile browser to check for errors?

**Comment by DanielnetoDotCom** [comment]
Is this failing on iphone and android?

can you please try this?

https://github.com/WWBN/AVideo/issues/9207#issuecomment-2226652032

**Comment by markginter31** [comment]
I don't have access to an iPhone to try.

I made those changes but still can't play mp4s.

You can see my test runs at https://video.oakhillcog.info

I'm really stumped.

**Comment by markginter31** [comment]
Also - the first couple of videos were fine (the ones at the beach).  It's
the recent ones I tried uploading that aren't working.

**Comment by DanielnetoDotCom** [comment]
Just tested this one on iPhone and works fine 

https://video.oakhillcog.info/channel/OakHillVideoChannel/video/5/sunrise-reprise

**Comment by DanielnetoDotCom** [comment]
I am not 100%, can you please try longer videos 1 minute or more?

**Comment by markginter31** [comment]
Just posted a 2 minute video.  MP4 won't work.  What's really weird is that
those sunrise videos uploaded and converted fine a month ago.... But any of
the newer videos don't (yet) 😀

**Comment by DanielnetoDotCom** [comment]
Weird, 

I uploaded your video and I got the same problem here 

https://demo.avideo.com/video/436/mp4-test-on-mobile


also I upload a different MP4 and there is no issue

https://demo.avideo.com/video/438/mp4-test-2 

so the issue is in the source file, there is probably something needs to be added on FFMPEG, but I am not sure what is in your mp4 video that is different from mine

If you want to inspect you can use those 2 files

## your file that fail to play on mobile

https://github.com/user-attachments/assets/5660b98d-d7c0-4612-a054-67a2715a3522

## a file that plays on mobile 

https://github.com/user-attachments/assets/6d0fbbe0-46e2-488d-87e3-0426ffb318da

**Comment by markginter31** [comment]
I figured it out.  At some point I set an option in my phone camera to
HDR10+  once these videos were uploaded and converted the video codec
showed as H.264 (High Profile 10) with a frame rate of 120.  The videos
that work showed video codec of H.264 (High Profile) with a frame rate of
24

I turned off the HDR10+ setting and now videos upload and encode fine.
Sorry for the hassle.  It was my own fault.
